src/ray/raylet/node_manager.cc

```diff
@@ -2766,6 +2769,91 @@ void NodeManager::SendSpilledObjectRestorationRequestToRemoteNode(
       });
 }
 
+void NodeManager::FillNormalTaskResourceUsage(
+    const std::shared_ptr<rpc::ResourcesData> &resources_data) {
+  auto new_resources = GetResourcesUsedByNormalTask();
+  if (new_resources) {
+    std::shared_ptr<std::unordered_map<std::string, double>> resources_changes;
+    if (nullptr == last_report_normal_task_resources_) {
+      resources_changes = new_resources;
+    } else {
+      resources_changes = GetNormalTaskResourcesChanges(
+          *last_report_normal_task_resources_, new_resources);
+    }
+
+    if (!resources_changes->empty()) {
+      for (const auto &iter : *resources_changes) {
+        (*resources_data->mutable_normal_task_resources_changes())[iter.first] =
+            iter.second;
+      }
+      last_report_normal_task_resources_ = new_resources;
+    }
+  }
+}
+
+std::shared_ptr<std::unordered_map<std::string, double>>
+NodeManager::GetResourcesUsedByNormalTask() const {
+  auto resources = std::make_shared<std::unordered_map<std::string, double>>();
+  if (!RayConfig::instance().new_scheduler_enabled()) {
+    return resources;
+  }
+
+  auto cluster_resource_scheduler =
+      std::dynamic_pointer_cast<ClusterResourceScheduler>(cluster_resource_scheduler_);
+  for (const auto &iter : leased_workers_) {
+    const auto &leased_worker = iter.second;
+    const auto &allocated_instances = leased_worker->GetAllocatedInstances();
+    if (leased_worker->GetActorId().IsNil() && allocated_instances != nullptr) {
+      const auto &predefined_resources = allocated_instances->predefined_resources;
+      for (size_t res_idx = 0; res_idx < predefined_resources.size(); res_idx++) {
+        for (size_t inst_idx = 0; inst_idx < predefined_resources[res_idx].size();
```

Comment:
Consider 
```
std::accumulate(predefined_resources[res_idx].begin(), predefined_resources[res_idx].end(), 0, std::plus<double>);
```

instead of this loop?

Reply:
Good idea, thx.